lib/rubocop/cop/rails/enum_hash.rb

```diff
@@ -30,6 +30,35 @@ def on_send(node)
             add_offense(node, message: format(MSG, enum: name))
           end
         end
+
+        def autocorrect(node)
+          enum_values = node.children[2].children.first.children[1]
+          to_replace = enum_values.loc.expression
+          values_hash = "{ #{converted_values(enum_values)} }"
+
+          ->(corrector) { corrector.replace(to_replace, values_hash) }
+        end
+
+        private
+
+        def converted_values(enum_values)
+          enum_values.children.each_with_index.map do |child, index|
+            hash_entry_as_string(child, index)
+          end.join(', ')
+        end
+
+        def hash_entry_as_string(child, index)
+          value = child.children.first
+          case value
+          when String
+            "'#{value}' => #{index}"
+          when Symbol
+            value = "'#{value}'" if value =~ /\s/
```

Comment:
I'd like to use a stricter way to quote the symbol. Comparing `/\s/` is not match `:'foo-bar'` and so on.

@koic Do you have any idea to make it strict? I guess we have a method to detect the necessity of quoting.

Reply:
@pocke I agree with it is better to be stricter, but unfortunately I have not got a good idea yet...

Reply:
https://github.com/rubocop-hq/rubocop/blob/971ac419d76e62d394a7abd9e941335574cb4dd3/lib/rubocop/cop/style/hash_syntax.rb#L140-L155

I found a check that `foo:` style is acceptable.
But it is not portable. We need to extend the method to a node extension or mixin if we'd like to use it for the cop.

So I think the current implementation is acceptable for now. I think it works on 99% cases because enum key name does not have special symbols in most cases.